src/language/parser.ts

```diff
@@ -1052,6 +1050,26 @@ export class Parser {
     });
   }
 
+  /**
+   * EnumValue : Name but not `true`, `false` or `null`
+   */
+  parseEnumValueName(): NameNode {
+    if (
+      this._lexer.token.value === 'true' ||
+      this._lexer.token.value === 'false' ||
+      this._lexer.token.value === 'null'
+    ) {
+      throw syntaxError(
+        this._lexer.source,
+        this._lexer.token.start,
+        `${getTokenDesc(
+          this._lexer.token,
+        )} is reserved and cannot be used for an enum value.`,
```

Comment:
BTW. Adding a similar message to fragment name parsing is welcomed if you want to work on such PR.